fastly/block_fastly_service_v1_backend.go

```diff
@@ -415,7 +420,8 @@ func (h *BackendServiceAttributeHandler) Register(s *schema.Resource) error {
 
 	s.Schema[h.GetKey()] = &schema.Schema{
 		Type:     schema.TypeSet,
-		Required: true,
+		Required: required,
+		Optional: !required,
```

Comment:
Two quick questions:

1. Do we need to provide _both_ fields, e.g. can't we just replace `Required` with `Optional`?
2. If we can just set `Optional`, then would the logic be simpler (and would it still work) if we reversed it?

e.g. 

```terraform
optional := false

if h.GetServiceMetadata().serviceType == ServiceTypeVCL {
  optional = true
  ...
}

s.Schema[h.GetKey()] = &schema.Schema{
  Optional: optional,
  ...
}
```

Reply:
I guess that works too. Let me test

Reply:
@Integralist Looks like TF core requires an explicit set as I did. Otherwise, error:

>        1 error occurred:
>        	* resource fastly_service_v1: backend: One of optional, required, or
>        computed must be set

so I leave that as-is.

Reply:
@smaeda-ks Ah ok, the error is still a bit weird to me because I wouldn't have expected there to have been an issue with the logic per se.